libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/Adapters.kt

```diff
@@ -42,6 +42,30 @@ class ListAdapter<T>(private val wrappedAdapter: Adapter<T>) : Adapter<List<@Jvm
   }
 }
 
+inline fun <reified T> arrayFromJson(wrappedAdapter: Adapter<T>, reader: JsonReader, customScalarAdapters: CustomScalarAdapters): Array<T> {
```

Comment:
Also interesting that inline functions are not considered by https://github.com/Kotlin/binary-compatibility-validator 👀 
I guess that makes sense because changes to inline function do not change the API but could still break source compatibility.

Reply:
Ah, went to make this change and remembered why I didn't make them private to begin with:

> Public-API inline function cannot access non-public-API

If I make `arrayFromJson` or `arrayToJson` private, the inline extension function won't be able to access them.

I guess that sort of makes sense.

The solution I found is to make them private member functions of the anonymous object. That will hide them from the public API and satisfies the compiler.

Reply:
Nice trick 👍